### PR TITLE
Adapt packaging scripts for maintenance branch

### PR DESCRIPTION
**Pull request by davidcassany**
## Why is this PR needed?

This is required in order to easily package skuba for v4 at any time.

## What does this PR do?

Includes a couple of packaging related backports from skuba master and it fixes the change log generation for v4.

## Anything else a reviewer needs to know?

Nothing special

## Info for QA

No changes on actual skuba code here.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
Didn't we remove the git build requirement?

**Comment by davidcassany** [comment]
> Didn't we remove the git build requirement?

Let me recheck, in fact the commit introducing it is just a two liner change and I was mostly interested in the other line...

**Comment by davidcassany** [comment]
> > Didn't we remove the git build requirement?
> 
> Let me recheck, in fact the commit introducing it is just a two liner change and I was mostly interested in the other line...

The git requierments is part of current master, in fact, it is since 70e0367b320bff5e4cc88e73c99db4a5415eb70a, which is the commit I cherry picked for v4. So the answer is no, we did not remove the git requirement.

File: ci/packaging/suse/rpmfiles_maker.sh [hunk 1]
```diff
@@ -37,30 +38,24 @@ then
...
     cp -v "$rpm_files/skuba.spec" "$rpm_files/skuba.tar.gz" "$work_dir/"
     cat "$rpm_files/skuba.changes.append" \
       "$work_dir/skuba.changes" \
       > "$tmp_dir/merged.changes" \
       && \
       cp -v "$tmp_dir/merged.changes" "$work_dir/skuba.changes"
+    log "Running OBS services"
+    (
+        cd "$work_dir"
+        osc -A 'https://api.suse.de' service disabledrun
```
[the remark this answers is not in the document]
Comment (davidcassany) [reply ci/packaging/suse/rpmfiles_maker.sh]: Oh, I see. correct, just fixed, thanks!

**Pull request merged**